### PR TITLE
Button fix

### PR DESCRIPTION
**Pull request by Henry-Tottle**

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -8,160 +50,117 @@ function shuffleArray(array) {
...
+                playGame(filmTitles, films)
+            } else {
+                endGame()
+            }
+        }, 200)
+        shuffleArray(filmTitles)
+    })
+}
 
 const playGame = (filmTitles, films) => {
```
Comment (mallowmew) [inline script.js:85]: Arguably another poorly-named function - is this playing a game or playing a round of the game?

File: script.js [hunk 2]
```diff
@@ -8,160 +50,117 @@ function shuffleArray(array) {
...
-
     playMenu.style.visibility = 'hidden'
-
-
-
+    quoteElem.classList.remove('hidden')
+    button1.classList.remove('hidden')
+    button2.classList.remove('hidden')
+    button3.classList.remove('hidden')
+    hintButton.classList.remove('hidden')
```
Comment (mallowmew) [inline script.js:91]: Two different ways to toggle element visibility right next to each other!

File: script.js [hunk 3]
```diff
@@ -1,5 +1,47 @@
+
 const playButton = document.querySelector('.play')
 const playMenu = document.querySelector('.gameButtons')
+const button1 = document.querySelector('.btn1')
+const button2 = document.querySelector('.btn2')
+const button3 = document.querySelector('.btn3')
```
Comment (mallowmew) [inline script.js:6]: Is it most appropriate to use classes or IDs for your buttons?

**Review by mallowmew — CHANGES_REQUESTED** [review]

File: script.js [hunk 4]
```diff
@@ -1,5 +1,47 @@
...
+        instructions.style.display = 'flex';
+        instructions.style.zIndex = '1';
+    } else {
+        instructions.style.display = 'none';
+    }
+}
+
 function shuffleArray(array) {
     for (var i = array.length - 1; i > 0; i--) {
         var j = Math.floor(Math.random() * (i + 1));
```
[the remark this answers is not in the document]
Comment (Henry-Tottle) [reply script.js:47]: stack overflow gem

File: script.js [hunk 5]
```diff
@@ -1,5 +1,47 @@
...
+}
+
+const modal = () => {
+    if (instructions.style.display === 'none') {
+        instructions.style.display = 'flex';
+        instructions.style.zIndex = '1';
+    } else {
+        instructions.style.display = 'none';
+    }
+}
```
Comment (mallowmew) [inline script.js:43]: This function is only called `modal`? Isn't that really confusing?

Do you know about F2 / Refactor to rename functions?

File: script.js (hunk 3 above)
Comment (Henry-Tottle) [reply script.js:6]: thank you :)

File: script.js [hunk 6]
```diff
@@ -1,5 +1,47 @@
...
+    const response = await fetch('./films.json')
+    const data = await response.json()
+    return data.films
+}
+
+const setupGame = async () => {
+    filmTitles = await fetchFilms()
+    films = await fetchFilms()
+    playButton.addEventListener('click', () => playGame(filmTitles, films))
+}
```
[the remark this answers is not in the document]
Comment (Henry-Tottle) [reply script.js:34]: stack overflow and Matt last minute yesterday, and friendly neighbourhood GPT

File: script.js [hunk 7]
```diff
@@ -8,160 +48,114 @@ function shuffleArray(array) {
...
 }
 
-resetBtn.addEventListener('click',resetGame)
-resetBtn.addEventListener('click',countDownTimer)
\ No newline at end of file
+const playGame = () => {
+    setupGame().
+    then(() => roundGenerator(filmTitles,films))
+    countDownTimer()
+}
```
Comment (mallowmew) [inline script.js:146]: So `setupGame` is going to go away and fetch the films, and then start the game eventually, after it has done that.
Once `setupGame` has started, execution is going to continue, and run the `countDownTimer` immediately.
So if our user has a slow connection, is that gong to be fair?
How do you ensure that the countdown only starts at the same time as the first round of the game?

File: script.js [hunk 8]
```diff
@@ -8,160 +48,114 @@ function shuffleArray(array) {
...
+    })
+}
 
+const roundGenerator = (filmTitles, films) => {
+    playMenu.classList.add('hidden')
+    quoteElem.classList.remove('hidden')
+    button1.classList.remove('hidden')
+    button2.classList.remove('hidden')
+    button3.classList.remove('hidden')
+    hintButton.classList.remove('hidden')
```
Comment (mallowmew) [inline script.js:88]: Do you need to un/hide everything every round?

**Review by mallowmew — CHANGES_REQUESTED** [review]

**Review by mallowmew — APPROVED** [review]

**Pull request merged**